### PR TITLE
created updateUser endpoint

### PR DESCRIPTION
**Pull request by iyervarsha**
This is the endpoint to update the user - for now it just updates the user type but in the future we could expand this endpoint to update the password and other fields

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -0,0 +1,2423 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
+# yarn lockfile v1
+
+
+"@babel/code-frame@^7.0.0":
```
Comment (michaelzhou01) [inline yarn.lock:5]: This is an npm project -- don't commit yarn.lock

File: src/router.ts [hunk 2]
```diff
@@ -13,4 +13,6 @@ router.get('/verifyuser', requireAuthUnverified, Auth.verifyUser);
 
 router.get('/user', requireAuth, User.getUser);
 
+router.put('/updateuser',requireAuth, User.updateUser);
```
Comment (michaelzhou01) [inline src/router.ts:16]: the route should be /user, put /updateuser is redundant

Comment (michaelzhou01) [reply src/router.ts:16]: consider using router.route to group endpoints

File: src/models/user_model.ts [hunk 3]
```diff
@@ -1,9 +1,22 @@
...
+        });
+    });
+};
+
+
+
 export const getUser = (email: string, verified = true): Promise<any> => {
     const qstr =
-        'SELECT email AS email, user.name AS name, password, user_type.name AS userType \
+        'SELECT email AS email, user.name AS name, password, user_type.id AS userType\
```
Comment (michaelzhou01) [inline src/models/user_model.ts:19]: email AS email is redundant

Comment (michaelzhou01) [reply src/models/user_model.ts:19]: We want user_type.name for the front end. We could 1) have subsequent queries find the user type id from the user type name 2) return user_type.id as well

File: src/controllers/user_controller.ts [hunk 4]
```diff
@@ -10,6 +10,25 @@ class BadRequestError extends Error {}
 
 dotenv.config();
 
+export const updateUser: RequestHandler = async (req, res) => {
+    User.getUser(req.body.email)
+        .then((userInfo) => {
+            if (userInfo === null) {
+                throw new BadRequestError('An account does not exist under this email.');
+            } else {
+                return User.updateUser(req.body.userType, userInfo.email);
```
Comment (michaelzhou01) [inline src/controllers/user_controller.ts:19]: userType is not passed to in the body. Only email is passed by the middleware as req.user. userType needs to be queried for

Comment (iyervarsha) [reply src/controllers/user_controller.ts:19]: it didn't work when I didn't use req.body.userType though. userInfo.userType just stores the previous userType

File: src/models/user_model.ts (hunk 3 above)
Comment (iyervarsha) [reply src/models/user_model.ts:19]: yeah think im going to select name and id

File: src/models/user_model.ts [hunk 5]
```diff
@@ -1,9 +1,22 @@
 import sql from '../config/dbpool';
 import bcryptjs from 'bcryptjs';
 
+export const updateUser = (userType: Number, email:string): Promise<any> => {
```
Comment (michaelzhou01) [inline src/models/user_model.ts:4]: We should have an overarching updateUser function that allows for updating of any user attribute.

File: yarn.lock (hunk 1 above)
Comment (iyervarsha) [reply yarn.lock:5]: ill add yarn.lock to the gitignore

File: src/controllers/user_controller.ts (hunk 4 above)
Comment (michaelzhou01) [reply src/controllers/user_controller.ts:19]: Oh wait my bad you're right 😅

File: src/models/user_model.ts [hunk 6]
```diff
@@ -1,9 +1,22 @@
...
+        });
+    });
+};
+
+
+
 export const getUser = (email: string, verified = true): Promise<any> => {
     const qstr =
-        'SELECT email AS email, user.name AS name, password, user_type.name AS userType \
+        'SELECT email, user.name AS name, password, user_type.id AS userType, user_type.name AS userTypeName\
```
Comment (michaelzhou01) [inline src/models/user_model.ts:19]: as UserTypeId

File: src/router.ts [hunk 7]
```diff
@@ -13,4 +13,6 @@ router.get('/verifyuser', requireAuthUnverified, Auth.verifyUser);
 
 router.get('/user', requireAuth, User.getUser);
 
+router.put('/user',requireAuth, User.updateUser);
```
Comment (michaelzhou01) [inline src/router.ts:16]: switch to router.route

File: yarn.lock (hunk 1 above)
Comment (michaelzhou01) [reply yarn.lock:5]: still need to remove this file from the commit

File: src/models/user_model.ts (hunk 5 above)
Comment (michaelzhou01) [reply src/models/user_model.ts:4]: See https://imgur.com/a/h0VWxIG

**Review by michaelzhou01 — APPROVED** [review]

**Pull request merged**